include/ajax.tickets.php

```diff
@@ -1273,7 +1273,7 @@ function markAs($tid, $action='') {
         if (!($ticket=Ticket::lookup($tid)))
             Http::response(404, __('No such ticket'));
 
-        if (!$ticket->checkStaffPerm($thisstaff, Ticket::PERM_REPLY) && !$thisstaff->isManager())
+            if (!$ticket->checkStaffPerm($thisstaff, Ticket::PERM_REPLY) && !$ticket->checkStaffPerm($thisstaff, Ticket::PERM_MARKANSWERED) && !$thisstaff->isManager())
```

Comment:
You should fix the spacing issue and change this to the following:
```
if (!$ticket->checkStaffPerm($thisstaff, Ticket:: PERM_MARKANSWERED) && !$thisstaff->isManager())
```

Since you are making this its own Permission there is no need to check wether or not they have Reply Permission, only the MarkAs Permission (and if they are Manager of course). The same concept as what you did with the `$canMarkAnswered` variable below.

Reply:
Correct, thanks!